Fix travis build.

maxalbert opened the pull request:
It looks like there was some change on Travis recently because [build #222](https://travis-ci.org/open-research/sumatra/builds/85697650) from a few weeks ago passed but [build #228](https://travis-ci.org/open-research/sumatra/builds/89987821) failed when the only difference was [the removal of two empty lines](https://github.com/open-research/sumatra/commit/3d5a3151380fea072838bae83d9d5a5f56d25e66).

This PR is an attempt to fix the Travis build. The `.travis.yml` file in this version is based on the example at http://conda.pydata.org/docs/travis.html. It doesn't use `requirements.txt` but relies on equivalent `environment.yml` files for conda environments instead. Not sure if this is the route @apdavison wants to go (we can easily change it), but the nice side effect is that it doesn't need separate files for the various supported Python versions (such as `requirements-2.6.txt`, `requirements-2.7.txt`, `requirements-3.4.txt`).

apdavison commented:
I'm happy to use `environment.yml`, but there is some problem with the configuration. If you look at the Travis logs it appears that Python 3.4 is being used for all runs, even those that are supposedly using 2.6 and 2.7.

maxalbert commented:
Good catch! I guess that's why it didn't need separate files for the individual Python versions. ;-) I have fixed this, but in the process also reverted back to the original method of using `requirements.txt` in combination with `pip`. Part of the reason is that I had some trouble to get the tests to pass with the `environments.yml` file for Python 2.6 (not really sure why, although I didn't spend a huge amount of time looking into it). The other reason is that it introduces a bit of overhead at the moment to maintain both `requirements.txt` and `environment.yml` and keep them in sync with not too much benefit (apart from maybe a slightly faster test run on Travis).

Let me know if you think the `environment.yml` files for conda would be useful, though, and I'm happy to put them back in.
